### PR TITLE
Add Info.plist DSL to Podspecs

### PR DESCRIPTION
**Pull request by amorde**
Progress towards https://github.com/CocoaPods/CocoaPods/issues/8753

### CODE REVIEW

File: lib/cocoapods-core/specification/consumer.rb [hunk 1]
```diff
@@ -353,6 +352,31 @@ def prepare_value(attr, value)
...
+      # Merges two values in a hash together based on the needs of the attribute
+      #
+      # @param  [Specification::DSL::Attribute] attr
+      #         the attribute for which that value is needed.
+      #
+      # @param  [Object] old the value from the original hash
+      #
+      # @param  [Object] new the value from the new hash
+      #
+      def merge_hash_value(attr, old, new)
```
Comment (dnkoutso) [inline lib/cocoapods-core/specification/consumer.rb]: Nit: add @return?

File: spec/specification/consumer_spec.rb [hunk 2]
```diff
@@ -180,6 +180,50 @@ module Pod
...
+          @subspec_consumer.info_plist.should == {
+            'SOME_VAR' => 'OTHER_VALUE',
+          }
+        end
+
+        it 'allows specifying values by platform' do
+          @spec.info_plist = { 'CFBundleIdentifier' => 'org.cocoapods.MyLib' }
+          @spec.osx.info_plist = { 'CFBundleIdentifier' => 'org.cocoapods.MyLibOSX' }
+          osx_consumer = Specification::Consumer.new(@spec, :osx)
+          osx_consumer.info_plist.should == { 'CFBundleIdentifier' => 'org.cocoapods.MyLibOSX' }
```
Comment (dnkoutso) [inline spec/specification/consumer_spec.rb]: Assert the other one as well?

File: lib/cocoapods-core/specification/consumer.rb [hunk 3]
```diff
@@ -353,6 +352,33 @@ def prepare_value(attr, value)
...
+      #
+      # @return [Object] the merged value
+      #
+      def merge_hash_value(attr, old, new)
+        case attr.name
+        when :info_plist
+          new
+        else
+          if new.is_a?(Array) || old.is_a?(Array)
+            r = [*old] + [*new]
```
Comment (segiddins) [inline lib/cocoapods-core/specification/consumer.rb]: `Array(old) + Array(new)`

**Comment by dnkoutso** [comment]
Any tests to be added for `json.rb`?

**Comment by amorde** [comment]
Added JSON tests and removed support for subspecs (but kept for app and test specs) based on discussion in RFC

**Pull request merged**